src/utilities/getMediaUrl.ts

```diff
@@ -0,0 +1,20 @@
+import { getClientSideURL } from '@/utilities/getURL'
+
+/**
+ * Processes media resource URL to ensure proper formatting
+ * @param url The original URL from the resource
+ * @param cacheTag Optional cache tag to append to the URL
+ * @returns Properly formatted URL with cache tag if provided
+ */
+export const getMediaUrl = (url: string | null | undefined, cacheTag?: string | null): string => {
+  if (!url) return ''
+
+  // Check if URL already has http/https protocol
+  if (url.startsWith('http://') || url.startsWith('https://')) {
```

Comment:
nit: can we use `new URL(url)` and look at the scheme using structured data, append query parameters using the methods, etc, instead of treating this like a string? I tried to do that in `middleware.ts` and I think (or hope?) that it's a more robust approach that makes fewer assumptions / is better-formed in the face of failures, cache tags that need url-encoding, etc.